storage/retry_conformance_test.go

```diff
@@ -683,12 +698,25 @@ func (et *emulatorTest) create(instructions map[string][]string) {
 	}
 
 	et.id = testRes.TestID
-
-	// Create wrapped client which will send emulator instructions
 	et.host.Path = ""
-	client, err := wrappedClient(et.T, et.id)
-	if err != nil {
-		et.Fatalf("creating wrapped client: %v", err)
+
+	// Create wrapped client which will send a retry test id via a constant header
+	var client *Client
+	if transport == "json" {
+		client, err = wrappedClient(et.T, et.id)
+		if err != nil {
+			et.Fatalf("creating wrapped client: %v", err)
+		}
+	} else if transport == "grpc" {
+		opts := defaultGRPCOptions()
+		header := metadata.New(map[string]string{"x-retry-test-id": et.id})
+		ctx := metadata.NewOutgoingContext(context.Background(), header)
+		var clientMu sync.Mutex
+		clientMu.Lock()
+		os.Setenv("STORAGE_USE_GRPC", "true")
+		client, _ = NewClient(ctx, opts...)
```

Comment:
We can use the new exported constructor after #8509 is merged. This will eliminate the need for the env var and mutex.

Reply:
Sounds good, thanks! I'll go ahead and merge the testbench implementation first since the proposed changes in this PR seems feasible; then I'll come back and finalize this :)